### ISSUE TITLE
Unclear how to resolve non-labelled hidden input fields

### ISSUE THREAD

**Issue by zackseuberling** [squizlabs/HTML_CodeSniffer#255]
I believe this is related to #80 and #107 but I'm not sure what the correct application is.

We're using hidden input fields and and populating them via Javascript for lat/lon location data:

```html
<input class="global-search__lat" step="0.01" type="number" name="lat" hidden="true">
<input class="global-search__lon" step="0.01" type="number" name="lon" hidden="true">
```

without a label, the sniffer alerts of an *error*: 

> This number input element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.

*with* a label, as such 

```html
<input class="global-search__lat" step="0.01" type="number" name="lat" hidden="true" aria-label="Latitude">
```

the sniffer alerts of a *warning*:

> This form field is intended to be hidden (using the "hidden" attribute), but is also labelled in some way. There should be no need to label a hidden form field.

I'm inclined to provide the mark-up that produces a warning, but I would like to resolve my markup so that the sniffer does not detect a problem at all. Is there an ideal solution for hidden elements without a label?

**Comment by ironikart** [comment]
This looks like a bug, I was able to replicate with your example html.

**Comment by zackseuberling** [comment]
thanks for looking into this! I really love the project, so thanks for all that you do :)

**Comment by ironikart** [comment]
I misread your initial test case, but I believe you need to use type="hidden" and not hidden="true" for your attributes. My understanding of the hidden attribute is that it visually hides the element but does not exclude it from screen readers. You can use `aria-hidden` for this which we do detect.

**Comment by zackseuberling** [comment]
thanks for looking into this. I guess it was just an HTML markup misunderstanding on my part. switching my markup to use `type=hidden` passed the accessibility requirements as expected
